server/bin/create_test_repos.py

```diff
@@ -0,0 +1,469 @@
+BuildArch:      noarch
+ 
+%%description
+This is a ${name} package.
+ 
+%%files
+ 
+%%changelog
+* Fri Mar 1 2019 Jiri Hnidek <jhnidek redhat com> 0
+- ${name}
+"""
+
+GPG_NAME_REAL = "candlepin"
+
+GPG_NAME_EMAIL = "noreply@candlepinproject.org"
+
+GPG_BATCH_GEN_SCRIPT_CONTENT = """
+Key-Type: 1
+Key-Length: 2048
+Subkey-Type: 1
+Subkey-Length: 2048
+Name-Real: %s
+Name-Email: %s
+Expire-Date: 0
+%%no-protection
+%%transient-key
+%%commit
+""" % (GPG_NAME_REAL, GPG_NAME_EMAIL)
+
+GPG_EXPORTED_CANDLEPIN_KEY = os.path.expanduser("~") + "/RPM-GPG-KEY-candlepin"
+
+RPMMACROS_CONTENT = """
+%%_signature gpg
+%%_gpg_name %s
+""" % GPG_NAME_REAL
+
+RPMBUILD_ROOT_DIR = os.path.expanduser("~") + "/rpmbuild"
+
+REPO_ROOT_DIR = "/var/lib/tomcat/webapps/ROOT"
+
+CANLDEPIN_SERVER_BASE_URL = "https://localhost:8443/candlepin/"
+
+CANDLEPIN_USER = 'admin'
+CANDLEPIN_PASS = 'admin'
+
+
+def run_command(command, verbose=False):
+    """
+    Try to run command in own process
+    """
+
+    # Run command in subprocess
+    process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+
+    # Print output of command
+    if verbose is True:
+        for line in process.stdout.readlines():
+            print line
+
+    # Wait for result of process
+    ret = process.wait()
+
+    return ret
+
+
+def read_test_data(filename):
+    """
+    Try to read test data from json file.
+    """
+    try:
+        with open(filename) as fp:
+            try:
+                data = json.load(fp)
+            except ValueError as err:
+                print("Unable to read file %s with test data: %s" % (filename, str(err)))
+                return None
+    except IOError as err:
+        print("Unable to read file: %s with test data: %s" % (filename, str(err)))
+        return None
+    return data
+
+
+def get_repo_definitions(test_data):
+    """
+    Get list of repositories from test_data
+    """
+
+    if test_data is None:
+        return []
+
+    try:
+        repo_definitions = test_data['content']
+    except KeyError as err:
+        print('Info: Test data does not include any global definition of repositories')
+        repo_definitions = []
+
+    # There can be also some content specific for owners
+    try:
+        owners = test_data['owners']
+    except KeyError as err:
+        return repo_definitions
+    
+    for owner in owners:
+        try:
+            owner_content = owner['content']
+        except KeyError as err:
+            continue
+        else:
+            repo_definitions.extend(owner_content)
+    
+    return repo_definitions
+
+
+def add_packages_to_repo(packages_list, repo_path, repository):
+    """
+    This function tries to add list of packages to the repository
+    """
+    for pkg_name in packages_list:
+        print("\tAdding package %s" % pkg_name)
+        pkg_file_name = pkg_name + '-1-0.noarch.rpm'
+        src_pkg_file_path = RPMBUILD_ROOT_DIR + '/RPMS/noarch/' + pkg_file_name
```

Comment:
Let's use `os.path.join` just for completeness sake.  I doubt this will ever be run on Windows, but seeing a raw concatenation of directory names makes my hair stand up on end for some reason.

Reply:
Good point.